solr/core/src/java/org/apache/solr/cloud/autoscaling/sim/SimClusterStateProvider.java

```diff
@@ -351,10 +340,10 @@ public void simSetClusterState(ClusterState initialState) throws Exception {
         // DocCollection will be created later
         collectionsStatesRef.put(dc.getName(), new CachedCollectionRef(dc.getName(), dc.getZNodeVersion()));
         collProperties.computeIfAbsent(dc.getName(), name -> new ConcurrentHashMap<>()).putAll(dc.getProperties());
-        opDelays.computeIfAbsent(dc.getName(), Utils.NEW_HASHMAP_FUN).putAll(defaultOpDelays);
+        opDelays.computeIfAbsent(dc.getName(), o -> new HashMap<String,Long>()).putAll(defaultOpDelays);
```

Comment:
This was added for a purpose.
`o -> new HashMap<String,Long>()` 

this creates a new lambda class. 
OTOH `Utils.NEW_HASHMAP_FUN` is created once and reused everywhere

Reply:
Is this some kind of tight inner loop where we need to be concerned about squeezing out every last gram of garbage? I'm not super familiar with this code, but it rather looks like a one-time initialization that will run once for each collection, so really it's a handful of objects we're talking about I think. Seems like a case of premature optimization, not really worth a warning from the compiler, IMO

Reply:
Should we be using Map.putIfAbsent instead of compute? The value mapped to the key doesn't actually use the value, so I don't believe we need to do computeIfAbsent, and if lambda functions are the issue, putIfAbsent will remove that altogether.

I'll try this out and run tests to see if that's an acceptable option, and report back in a bit.

Reply:
`putIfAbsent()` is wore than `computeIfAbsent()` 

the former actually create an object even before the method is called. The latter does it only after if the key has no value